pkg/controller/install/deployment.go

```diff
@@ -84,7 +85,22 @@ func NewStrategyDeploymentInstaller(strategyClient wrappers.InstallStrategyDeplo
 }
 
 func (i *StrategyDeploymentInstaller) installDeployments(deps []v1alpha1.StrategyDeploymentSpec) error {
-	for _, d := range deps {
+	for n, d := range deps {
+		if n > 0 {
+			// Check previous deployment
+			prev := deps[n-1]
+			ready, err := i.strategyClient.IsDeploymentReady(i.owner.GetNamespace(), prev.Name)
+			if err != nil {
+				log.Infof("Unable to retrieve deployment: %v", err)
+				return err
+			}
+
+			if !ready {
+				msg := fmt.Sprintf("Waiting on deployment %s/%s to be ready", prev.Name, i.owner.GetNamespace())
+				log.Infof(msg)
+				return fmt.Errorf(msg)
```

Comment:
1. what happens if the previous deployment isn't ready? I thought we are expected to wait rather than error out? 
2. Is this expect every loop to instantly succeed? I did not see any wait function in the loop. 
Correct me if I'm wrong. Thanks!

Reply:
> what happens if the previous deployment isn't ready?

Then the next deployment is not installed until it is ready. The returned error simply requeues the CSV in the installing phase.

Reply:
ahh, requeue. Thanks.